staking-pool/src/lib.rs

```diff
@@ -127,8 +154,73 @@ impl StakingContract {
         let contract: StakingContract = env::state_read().expect("ERR_READ_CONTRACT_STATE");
         contract
     }
+
+    /// create or update tier min point config
+    pub fn set_tier_point(&mut self, tier: Tier, min_point: U64) {
+        self.config.set_tier_min_point_cfg(tier, min_point as u64);
+    }
+
+    /// Get user point(xKula) amount by account id
+    pub fn get_user_point(&self, account_id: AccountId) -> U64 {
+        let account: Option<UpgradableAccount> = self.accounts.get(&account_id);
+        if account.is_some() {
+            let acc: Account = Account::from(account.unwrap());
+            U128(acc.point) as U64
+        } else {
+            U64(0)
+        }
+    }
+
+    /// Get user staking tier and point by account id
+    pub fn get_user_staking_tier(&self, account_id: AccountId) -> (Tier, U64) {
+        let point = self.get_user_point(account_id);
+        let mut user_tier = Tier::Tier0;
+        for (tier, min_point) in self.config.tier_point_config.to_vec() {
+            if point >= min_point {
+                user_tier = tier
+            } else {
+                break
+            }
+        }
+
+        (user_tier, point)
+    }
+
+    /// Get user staking tier trimming down:
+    /// Use for calculating Ticket & Allocation
+    /// Eg: 9.999 point =
+    ///     5.000 + 4.000 + 999
+    ///     1 x T3 = 100
+    ///     4 x T2 = 48
+    ///     9 x T1 = 9
+    ///     100 + 48 + 9 = 157
+    ///
+    /// return Vec<(Tier, min_tier_point, total_valid_point_at_this_tier)>
+    pub fn get_user_staking_matched_tiers(&self, mut point: U64) -> Vec<(Tier, U64, U64)> {
```

Comment:
Should we convert the tuple of (Tier, U64, U64) into a struct in order to easily use it on FE? 
The function name should be _get_matched_tiers_. Because the function doesn't depend on any user.